types/database/localDatabase.types.ts

```diff
@@ -3,6 +3,10 @@ import {SQLiteDatabase} from 'react-native-sqlite-storage';
 interface UseLocalDatabaseHook {
   localDb: SQLiteDatabase;
   refresh: (key: 'channelList' | 'channelInfo' | 'chat' | 'user' | 'channelMember') => void;
+  refreshWithId: (
+    key: 'channelList' | 'channelInfo' | 'chat' | 'user' | 'channelMember',
+    id: string
+  ) => void;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of the `refreshWithId` function is a good improvement for targeted data refreshing. However, it would be better to use an enum instead of string literals for the `key` parameter. This will help prevent potential typos and improve code maintainability.

```diff
- refreshWithId: (
-   key: 'channelList' | 'channelInfo' | 'chat' | 'user' | 'channelMember',
-   id: string
- ) => void;

+ refreshWithId: (
+   key: DatabaseKey,
+   id: string
+ ) => void;
```
Where `DatabaseKey` is an enum defined as:

```typescript
enum DatabaseKey {
  ChannelList = 'channelList',
  ChannelInfo = 'channelInfo',
  Chat = 'chat',
  User = 'user',
  ChannelMember = 'channelMember'
}
```
This change will make your code more robust and less prone to errors due to mistyped strings.


<!-- This is an auto-generated comment by OSS CodeRabbit -->